encoding/cloner.go

```diff
@@ -0,0 +1,29 @@
+package encoding
+
+import (
+	refmt "github.com/polydawn/refmt"
+	"github.com/polydawn/refmt/obj/atlas"
+)
+
+// PooledCloner is a thread-safe pooled object cloner.
+type PooledCloner struct {
+	Count   int
+	cloners chan refmt.Cloner
+}
+
+// SetAtlas set sets the pool's atlas. It is *not* safe to call this
+// concurrently.
```

Comment:
If it's not safe to call concurrently, make it only possible to do at construction time?_

Reply:
Unfortunately, we need to update the atlas each time the user registers a new type (which must happen at init). We could lock everything in place with a sync.Once on first use but that'd add a bit of overhead and I'm not sure if it's worth it.

Reply:
So make the caller register all types before starting to use it, and if they change their mind, they have to make a new instance of the whole pool.

I struggle with how else I'd use this in a semantically race-free way short of either A) doing the all-setup-before-use flow anyway, or B) mutexing the whole pool.